feat(layout): customized rows and gaps (VIV-1046) (#1143)
Vonage/vivid-3: libs/components/src/lib/layout/README.md

</vwc-layout>
```

### Row Spacing

> Use the `row-spacing` attribute to set the initial preferred `spacing` of a row.

Reviewer: ```suggestion
Use the `row-spacing` attribute to set the initial preferred margin between rows.
```
Can you also change this for column spacing please?

Author: it is not margin - it is spacing, or better yet - its `gap`

Control the grid-template-columns of the layout by setting `--layout-grid-template-columns`.

- Type: `string`
- Default: `undefined`

```html preview full
<vwc-layout style="--layout-grid-template-columns: 1fr 1fr;">
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
</vwc-layout>
```

### Grid-template-rows

Control the grid-template-rows of the layout by setting `--layout-grid-template-rows`.

> - Type: `string`

Reviewer: `type` should be removed from CSS Variables methinks. Makes no sense IMO.
@olaf-k , WDYT?

Reviewer 2: I don't think it should be `string`, because it's not, nor entirely removed, because it is useful to know what you can use.
a simple link to https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-rows would do the job.

Author: @olaf-k , its a good idea, I'll add the link, 
but I think @YonatanKra is asking in general about our documentation. 
We have "type" to all the places where we have css variable.
Maybe this is a discussion for a separate task :)

Author: https://jira.vonage.com/browse/VIV-1057

Reviewer: I agree with @olaf-k regarding the link. I believe I offered it on Slack for another place where we assume knowledge (I believe it was regarding `auto-size` or something).

If we "imitate" CSS standards this way, we should link from professional jargon to some external documentation. 

I certainly don't know these things out of my head and need to google them :)

- Default: `undefined`

```html preview full
<vwc-layout style="--layout-grid-template-rows: 80px 40px auto;">
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
    <vwc-card elevation="2"></vwc-card>
  <vwc-card elevation="2"></vwc-card>
</vwc-layout>
```

### Grid-gap

> #### Grid-template-row-gap

Reviewer: Shouldn't this be the name of the `css variable`? 
```suggestion
#### --layout-column-gap
```
I know it's not the same in other components, but I believe this makes it clearer.
Anyway, in other components, the title is without the `-` between words.
I think just writing the actual property is the best, as it makes things clearer. Or at least write it as a title without the `-` so it will not confuse ppl (like me).
